### ISSUE TITLE
How does this algorithm perform temporal prediction?\该算法如何进行时序预测？

### ISSUE THREAD

**Issue by Mr-White-W** [Frank-Wang-oss/FCSTGNN#2]
Dear experts and scholars, hello! I would like to ask a question. This article uses fully connected graph networks to extract features from different sensors and different time points. I am wondering how these features are ultimately used for time series data prediction. As a newcomer to time series prediction, I have noticed that many papers utilize RNN-type networks for time feature extraction or time series prediction. However, I do not fully understand how the authors achieve the final prediction. I would appreciate it if you could provide me with some clarification.
尊敬的专家学者，您好！我在研究一篇文章，该文章利用全连接图网络从不同传感器和不同时间点中提取特征。我很想知道最后是如何利用这些特征进行时序数据预测的。作为时序预测领域的新手，我发现许多论文都采用了RNN类型的网络来进行时间特征提取或时序预测。但我并不完全理解作者是如何实现最终预测的。希望各位能够为我解答这个问题。

**Comment by Frank-Wang-oss** [comment]
Your question can be solved by the code for RUL prediction. Specifically, for prediction, we just simply use the extracted features and then map the features to a 1-dimensional vector by a linear projection nn.Linear(feature_dimension, 1).

**Comment by Mr-White-W** [comment]
> 您的问题可以通过 RUL 预测代码来解决。具体来说，对于预测，我们只是简单地使用提取的特征，然后通过线性投影 nn 将特征映射到一维向量。线性（feature_dimension， 1）.

Thank you for your reply. What you mean is that the method in your paper mainly solves the problem of feature extraction, but does not solve the prediction problem. So should your comparative experiment only focus on different spatiotemporal feature extraction algorithms? Another question is about the time it takes for you to train the model? I'm sorry, I couldn't find the appendix document you mentioned in the article, and I'm not sure about the specific time it takes for your algorithm's training and actual calling process.
感谢您的回复。您的意思是指，您论文中的方法主要解决的是特征提取问题，并没有解决预测问题。所以您的对比实验应该只是针对不同时空特征提取算法吗？还有一个问题是关于您训练模型所花费的时间？很抱歉，我未能找到您在文章中提到的appendix文档，不清楚您的算法的训练与实际调用过程具体耗费的时间。

**Comment by Mr-White-W** [comment]
> Thank you for your reply. What you mean is that the method in your paper mainly solves the problem of feature extraction, but does not solve the prediction problem. So should your comparative experiment only focus on different spatiotemporal feature extraction algorithms? Another question is about the time it takes for you to train the model? I'm sorry, I couldn't find the appendix document you mentioned in the article, and I'm not sure about the specific time it takes for your algorithm's training and actual calling process.
> 感谢您的回复。您的意思是指，您论文中的方法主要解决的是特征提取问题，并没有解决预测问题。所以您的对比实验应该只是针对不同时空特征提取算法吗？还有一个问题是关于您训练模型所花费的时间？很抱歉，我未能找到您在文章中提到的appendix文档，不清楚您的算法的训练与实际调用过程具体耗费的时间以及您所采用的设备。



> 您的问题可以通过 RUL 预测代码来解决。具体来说，对于预测，我们只是简单地使用提取的特征，然后通过线性投影 nn 将特征映射到一维向量。线性（feature_dimension， 1）.

Thank you for your reply. What you mean is that the method in your paper mainly solves the problem of feature extraction, but does not solve the prediction problem. So should your comparative experiment only focus on different spatiotemporal feature extraction algorithms? Another question is about the time it takes for you to train the model? I'm sorry, I couldn't find the appendix document you mentioned in the article. I'm not sure about the specific time it takes to train and actually call your algorithm, as well as the device you used.
感谢您的回复。您的意思是指，您论文中的方法主要解决的是特征提取问题，并没有解决预测问题。所以您的对比实验应该只是针对不同时空特征提取算法吗？还有一个问题是关于您训练模型所花费的时间？很抱歉，我未能找到您在文章中提到的appendix文档，不清楚您的算法的训练与实际调用过程具体耗费的时间以及您所采用的设备。

**Comment by Frank-Wang-oss** [comment]
Yes this work is for feature extraction of MTS data. Not just for prediction tasks, but it can be used for prediction tasks. 
Regarding computation time, we have updated in our latest version but we have not updated in Arxiv. We will update our Arxiv ASAP.
